### PR TITLE
MCOL-3357

### PR DESCRIPTION
**Pull request by benthompson15**

### CODE REVIEW

File: storage-manager/src/Cache.cpp [hunk 1]
```diff
@@ -56,20 +56,32 @@ Cache::Cache()
...
-        logger->log(LOG_CRIT, "Cache/cache_size is not set");
-        throw runtime_error("Please set Cache/cache_size in the storagemanager.cnf file");
+        logger->log(LOG_CRIT, "Cache/cache_size is not set. Using default value = %zi",defaultCacheValue);
+        maxCacheSize = defaultCacheValue;
+        //throw runtime_error("Please set Cache/cache_size in the storagemanager.cnf file");
     }
     try
     {
-        maxCacheSize = stoul(stmp);
+        size_t configCacheValue = stoul(stmp);
```
Comment (pleblanc1976) [inline storage-manager/src/Cache.cpp:69]: stoul() only handles unsigned 64-bit, might want to make that match the size_t type.

**Review by pleblanc1976 — CHANGES_REQUESTED** [review]
Doesn't implement the listener interface.  Should be mainly a cut & paste job to make it do that.  Listener pattern -> better encapsulation & more maintainable.

File: storage-manager/src/Cache.cpp (hunk 1 above)
Comment (LinuxJedi) [reply storage-manager/src/Cache.cpp:69]: Did you mean unsigned 32-bit? stoull() is the 64-bit one.

Comment (pleblanc1976) [reply storage-manager/src/Cache.cpp:69]: Sry, typo, yes, stoull() is what you're after Ben.

**Pull request closed without merging**